elytron/src/main/java/org/wildfly/extension/elytron/HttpAuthenticationFactoryAddHandler.java

```diff
@@ -0,0 +1,97 @@
+
+
+/**
+ * A {@link OperationStepHandler} for adding a {@link HttpAuthenticationFactoryService} for an HttpAuthenticationFactory.
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+
+abstract class HttpAuthenticationFactoryAddHandler<T> extends BaseAddHandler {
+
+    private final RuntimeCapability<?> runtimeCapability;
+    private final ServiceController.Mode initialMode;
+    private final ServiceController.Mode embeddedInitialMode;
+
+    HttpAuthenticationFactoryAddHandler(Class<T> serviceType, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
+        this(serviceType, ServiceController.Mode.ACTIVE, attributes, runtimeCapability);
+    }
+
+    HttpAuthenticationFactoryAddHandler(Class<T> serviceType, ServiceController.Mode initialMode, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
+        this(serviceType, initialMode, initialMode, attributes, runtimeCapability);
+    }
+
+    HttpAuthenticationFactoryAddHandler(Class<T> serviceType, ServiceController.Mode initialMode, ServiceController.Mode embeddedInitialMode, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
+        super(new HashSet<>(Collections.singletonList(checkNotNullParam("runtimeCapabilities", runtimeCapability))), attributes);
+        this.runtimeCapability = runtimeCapability;
+        checkNotNullParam("serviceType", serviceType);
+        this.initialMode = checkNotNullParam("initialMode", initialMode);
+        this.embeddedInitialMode = checkNotNullParam("embeddedInitialMode", embeddedInitialMode);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected final void performRuntime(OperationContext context, ModelNode operation, Resource resource)
+            throws OperationFailedException {
+        HttpAuthenticationFactoryService httpAuthenticationFactoryService = new HttpAuthenticationFactoryService();
+
+        ServiceBuilder<T> serviceBuilder = (ServiceBuilder<T>)context.getCapabilityServiceTarget().addCapability(runtimeCapability);
+        serviceBuilder.setInstance(httpAuthenticationFactoryService);
+
+        httpAuthenticationFactoryService.setHttpAuthenticationFactorySupplier(getHttpAuthenticationFactorySupplier(serviceBuilder, context, resource.getModel()));
+
+        installedForResource(commonDependencies(serviceBuilder, dependOnProperties(), dependOnProviderRegistration())
+                .setInitialMode(context.getProcessType() == ProcessType.EMBEDDED_SERVER && context.getRunningMode() == RunningMode.ADMIN_ONLY ? embeddedInitialMode : initialMode)
+                .install(), resource);
+    }
+
+    protected boolean dependOnProperties() {
+        return true;
+    }
+
+    protected boolean dependOnProviderRegistration() {
+        return true;
+    }
+
+    protected void installedForResource(ServiceController<T> serviceController, Resource resource) {}
```

Comment:
This method should be removed and the call to it removed, since it is a no-op method and there won't be any subclasses of this overriding it to do something.